Projects/Server/Gumps/New/Components/GumpBuilder.cs

```diff
@@ -0,0 +1,698 @@
+        {
+            ref readonly StaticStringsHandler stringsWriter = ref builder.StringsWriter;
+
+            builder.FinalizeLayout();
+
+            int worstLayoutLength = Zlib.MaxPackSize(builder.LayoutSize);
+            int worstStringsLength = Zlib.MaxPackSize(stringsWriter.BytesWritten);
+
+            SpanWriter compressedLayoutWriter = new(int.Max(worstLayoutLength, worstStringsLength));
+            OutgoingGumpPackets.WritePacked(builder.Layout, ref compressedLayoutWriter);
+
+            layout = new(compressedLayoutWriter.Span.ToArray(), builder.LayoutSize - 1);
+            strings = builder.StringsWriter.Finalize(compressedLayoutWriter.RawBuffer);
+
+            compressedLayoutWriter.Dispose();
+        }
+
+        public static void AddHtmlSlot(this ref GumpBuilder<DynamicStringsHandler> builder, int x, int y, int width, int height,
+            string slotKey, bool background = false, bool scrollbar = false)
+        {
+            DynamicStringsHandler.DynamicMode = true;
+            builder.AddHtml(x, y, width, height, $"key_{slotKey}", background, scrollbar);
+            DynamicStringsHandler.DynamicMode = false;
+        }
+
+        public static void AddLabelSlot(this ref GumpBuilder<DynamicStringsHandler> builder, int x, int y, int hue, ReadOnlySpan<char> slotKey)
+        {
+            DynamicStringsHandler.DynamicMode = true;
+            builder.AddLabel(x, y, hue, $"key_{slotKey}");
+            DynamicStringsHandler.DynamicMode = false;
+        }
+
+        public static void AddLabelCroppedSlot(this ref GumpBuilder<DynamicStringsHandler> builder, int x, int y,
+            int width, int height, int hue, string slotKey)
+        {
+            DynamicStringsHandler.DynamicMode = true;
+            builder.AddLabelCropped(x, y, width, height, hue, $"key_{slotKey}");
+            DynamicStringsHandler.DynamicMode = false;
+        }
+
+        public static void AddTextEntrySlot(this ref GumpBuilder<DynamicStringsHandler> builder, int x, int y,
+            int width, int height, int hue, int entryId, string slotKey)
+        {
+            DynamicStringsHandler.DynamicMode = true;
+            builder.AddTextEntry(x, y, width, height, hue, entryId, $"key_{slotKey}");
+            DynamicStringsHandler.DynamicMode = false;
+        }
+
+        public static void AddTextEntryLimitedSlot(this ref GumpBuilder<DynamicStringsHandler> builder, int x, int y,
+            int width, int height, int hue, int entryId, ReadOnlySpan<char> slotKey, int size = 0)
+        {
+            DynamicStringsHandler.DynamicMode = true;
+            builder.AddTextEntryLimited(x, y, width, height, hue, entryId, $"key_{slotKey}", size);
+            DynamicStringsHandler.DynamicMode = false;
+        }
+    }
+
+    static file class Properties
+    {
+        public static readonly byte[] Buffer = Encoding.ASCII.GetBytes("{ nomove }{ noclose }{ nodispose }{ noresize }");
```

Comment:
Let's also switch this to u8's like the others.

Reply:
I forgot to delete the old buffer, it it not used anymore: I just deleted it